docs/01_nodeos/06_logging/20_third_party_logging/10_deep_mind_logger.md

```diff
@@ -0,0 +1,42 @@
+---
+content_title: Deep-mind Logger Integration
+link_text: Deep-mind Logger Integration
+---
+
+## Overview
+
+The `Deep-mind logger` is part of the `dfuse` [platform]([https://dfuse.io/](https://dfuse.io/)) which is a highly scalable and performant [open-source]([https://github.com/dfuse-io/dfuse-eosio/tree/master](https://github.com/dfuse-io/dfuse-eosio/tree/master)) platform for searching and processing blockchain data.
+
+### How To Enable Deep-mind Logger
+
+EOSIO 3.0 integrates the `nodeos` core service daemon with `deep-mind logger`. To benefit from full `deep-mind` logging functionality you must start your `nodeos` instance with the flag `--deep-mind`. After the start you can observe in the `nodeos` console output the informative details outputs created by the `deep-mind` logger. They distinguish themselves from the default `nodeos` output lines because they start with the `DMLOG` keyword. \
+ \
+Example of `deep-mind` log lines:
+
+```sh
+DMLOG START_BLOCK 30515
+
+DMLOG TRX_OP CREATE onblock 308f77bf49ab4ddde74d37c7310c0742e253319d9da57ebe51eb7b35f1ffe174 {"expiration":"2020-11-12T10:13:06","ref_block_num":30514,...}
+
+DMLOG CREATION_OP ROOT 0
+
+DMLOG RLIMIT_OP ACCOUNT_USAGE UPD {"owner":"eosio","net_usage":{"last_ordinal":1316982371,"value_ex":0,"consumed":0},"cpu_usage":{"last_ordinal":1316982371,"value_ex":24855,"consumed":101},"ram_usage":27083}
+
+DMLOG APPLIED_TRANSACTION 30515 {"id":"308f77bf49ab4ddde74d37c7310c0742e253319d9da57ebe51eb7b35f1ffe174","block_num":30515,"block_time":"2020-11-12T10:13:05.500",...}
+
+DMLOG RLIMIT_OP STATE UPD {"average_block_net_usage":{"last_ordinal":30514,"value_ex":0,"consumed":0},"average_block_cpu_usage":{"last_ordinal":30514,...} \
+DMLOG ACCEPTED_BLOCK 30516 {"block_num":30516,"dpos_proposed_irreversible_blocknum":30516,"dpos_irreversible_blocknum":30515,...
+```
```

Comment:
For more clarity It's better to split actual shell commands from output:

For instance:
\`\`\`sh
DMLOG RLIMIT_OP ACCOUNT_USAGE UPD {"owner":"eosio","net_usage":{"last_ordinal":1316982371,"value_ex":0,"consumed":0},"cpu_usage":{"last_ordinal":1316982371,"value_ex":24855,"consumed":101},"ram_usage":27083}
\`\`\`

becomes:

\`\`\`sh
DMLOG RLIMIT_OP ACCOUNT_USAGE UPD {"owner":"eosio","net_usage":
\`\`\`
\`\`\`console (or \`\`\`json if output is JSON as in this case)
{"last_ordinal":1316982371,"value_ex":0,"consumed":0},"cpu_usage":{"last_ordinal":1316982371,"value_ex":24855,"consumed":101},"ram_usage":27083}
\`\`\`

etc...

Reply:
everything you see here is just shell output from nodeos, no shall command

Reply:
I have updated the text preceding these examples to be clear that those are coming from nodeos console output.

Reply:
Ah got it! In that case it might be wise to rename the codeblock to \`\`\`console (or remove 'sh') since \`\`\`sh is reserved for actual typed shell commands - eventually the docsgen tool might prepend shell prompts $ to each new line in \`\`\`sh codeblocks across the dev portal, which would then render incorrectly in this page.